pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/fwd/CLPForwardIndexCreatorV2.java

```diff
@@ -164,18 +180,29 @@ public CLPForwardIndexCreatorV2(File baseIndexDir, ColumnStatistics columnStatis
    *
    * @param baseIndexDir The base directory where the forward index files will be stored.
    * @param clpMutableForwardIndex The mutable forward index containing the raw data to be ingested.
-   * @param chunkCompressionType The compression type to be used for encoding the forward index.
+   * @param chunkCompressionType The compression type used for encoding the forward index.
+   * @param forceRawEncoding If true, raw bytes encoding will be used, bypassing CLP encoding.
    * @throws IOException If there is an error during initialization or while accessing the file system.
    */
   public CLPForwardIndexCreatorV2(File baseIndexDir, CLPMutableForwardIndexV2 clpMutableForwardIndex,
-      ChunkCompressionType chunkCompressionType)
+      ChunkCompressionType chunkCompressionType, boolean forceRawEncoding)
       throws IOException {
     _chunkCompressionType = chunkCompressionType;
 
     // Pick up metadata from mutable forward index and use it to initialize the immutable forward index
     _column = clpMutableForwardIndex.getColumnName();
     _numDoc = clpMutableForwardIndex.getNumDoc();
-    _isClpEncoded = clpMutableForwardIndex.isClpEncoded();
+
+    _intermediateFilesDir =
```

Comment:
will we cleanup the intermediateFilesDir ?

Reply:
Yes, it is cleaned up in the `close()` method
```
 @Override
  public void close()
      throws IOException {
    // Delete all temp files
    FileUtils.deleteDirectory(_intermediateFilesDir);   <-  delete directory
    _dataFile.close();
  }
```